x-pack/filebeat/input/httpjson/internal/v2/input.go

```diff
@@ -0,0 +1,225 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package v2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	retryablehttp "github.com/hashicorp/go-retryablehttp"
+	"go.uber.org/zap"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	inputcursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+	"github.com/elastic/beats/v7/libbeat/common/useragent"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/go-concert/ctxtool"
+	"github.com/elastic/go-concert/timed"
+)
+
+const (
+	inputName = "httpjsonv2"
+)
+
+var (
+	userAgent = useragent.UserAgent("Filebeat")
+
+	// for testing
+	timeNow = time.Now
+)
+
+type retryLogger struct {
+	log *logp.Logger
+}
+
+func newRetryLogger() *retryLogger {
+	return &retryLogger{
+		log: logp.NewLogger("httpjson.retryablehttp", zap.AddCallerSkip(1)),
```

Comment:
I recommend using the `inputName` constant here to construct the name. Or construct the logger from the parent logger (`parentLogger.Named("retryablehttp")`) and the new name will automatically inherit from the parent loggers name (it will join them with a dot as in parent_name + "." + child_name) plus it will inherit any structured data assigned to the logger.